target/linux/layerscape/patches-5.4/802-can-0025-can-flexcan-add-LPSR-mode-support-for-i.MX7D.patch

```diff
@@ -69,17 +68,17 @@ Signed-off-by: Joakim Zhang <qiangqing.zhang@nxp.com>
  
  	priv->can.state = CAN_STATE_ERROR_ACTIVE;
  	if (netif_running(dev)) {
-@@ -1984,15 +1987,19 @@ static int __maybe_unused flexcan_resume
+@@ -1982,11 +1987,19 @@ static int __maybe_unused flexcan_resume
  			if (err)
  				return err;
  		} else {
+-			err = flexcan_chip_enable(priv);
 +			pinctrl_pm_select_default_state(device);
 +
- 			err = pm_runtime_force_resume(device);
- 			if (err)
- 				return err;
- 
--			err = flexcan_chip_enable(priv);
++			err = pm_runtime_force_resume(device);
```

Comment:
It's same here. The two functions should not be needed here any more.

pinctrl_pm_select_default_state(device);
pm_runtime_force_resume(device);

Reply:
It can be further simplified to
`--- a/drivers/net/can/flexcan.c                                                  
+++ b/drivers/net/can/flexcan.c                                                  
@@ -26,6 +26,7 @@                                                                
 #include <linux/platform_device.h>                                              
 #include <linux/pm_runtime.h>                                                   
 #include <linux/regulator/consumer.h>                                           
+#include <linux/pinctrl/consumer.h>                                             
 #include <linux/regmap.h>                                                       
                                                                                 
 #define DRV_NAME                       "flexcan"                                
@@ -1954,9 +1955,7 @@ static int __maybe_unused flexcan_suspen                   
                        if (err)                                                 
                                return err;                                      
                } else {                                                         
-                       err = flexcan_chip_disable(priv);                        
-                       if (err)                                                 
-                               return err;                                      
+                       flexcan_chip_stop(dev);                                  
                }                                                                
                netif_stop_queue(dev);                                           
                netif_device_detach(dev);                                        
@@ -1982,7 +1981,9 @@ static int __maybe_unused flexcan_resume                   
                        if (err)                                                 
                                return err;                                      
                } else {                                                         
-                       err = flexcan_chip_enable(priv);                         
+                       err = flexcan_chip_start(dev);                           
+                       if (err)                                                 
+                               return err;                                      
                }                                                                
        }                                                                        
    `

Reply:
@yangbolu1991 - Thanks for the review and for the new patch; test building now.
@ldir-EDB0 - OK... I am test building with your modification of yang's modification now.